<launch>

    <arg name="stargazer_config" default="$(find landmark_finder)/res/stargazer.yaml"/>
    <arg name="undistorted_image_topic" default="/stellar/ir/image_raw"/>
    <arg name="landmark_topic" default="/landmarks_seen"/>
    <arg name="debug_mode" default="true"/>

    <node pkg="landmark_finder" type="landmark_finder_node" name="landmark_finder_node" output="screen">
        <param name="stargazer_config" value="$(arg stargazer_config)"/>
        <param name="landmark_topic" value="$(arg landmark_topic)"/>
        <param name="undistorted_image_topic" value="$(arg undistorted_image_topic)"/>

        <param name="threshold" value="100"/>
        <param name="tight_filter_size" value="3"/>
        <param name="wide_filter_size" value="31"/>
        <param name="maxRadiusForPixelCluster" value="3"/>
        <param name="minPixelForCluster" value="1"/>
        <param name="maxPixelForCluster" value="1000"/>
        <param name="maxRadiusForCluster" value="40"/>
        <param name="minPointsPerLandmark" value="5"/>
        <param name="maxPointsPerLandmark" value="9"/>
        <param name="debug_mode" value="$(arg debug_mode)"/>
    </node>

</launch>